### PR TITLE
Fixed NPE in Staff Portal when loading VT2 images

### PR DESCRIPTION
**Pull request by KDNeufeld**
# Description

This PR includes the following proposed change(s):

Fixed NPE in Staff Portal

The Staff Portal would not load VT2 images due to a mismatch of properties, namely the `act_or_regulation_name_code` field is only on VT1 images, not VT2. There is no such property on VT2 images called `act_or_regulation_name_code` and so this field may be undefined.

Note: I noticed that the Staff Portal may try to use this field for determining if the Count sections are valid or not (presumbably to pull the "MVA" text for the act portion of the Act/Sect/Desc fields. Currently the portal thinks the counts are always invalid.

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Refactoring / Documentation
- [ ] Version change

if your change is a breaking change, please add `breaking change` label to this PR

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

## Does the change impact or break the Docker build?

- [ ] Yes
- [x] No

If Yes: Has Docker been updated accordingly?

- [ ] Yes
- [ ] No

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: src/frontend/staff-portal/src/app/services/violation-ticket.service.ts [hunk 1]
```diff
@@ -129,7 +129,7 @@ export class ViolationTicketService implements IViolationTicketService {
         const foundViolationTicketCount2 = violationTicket.violationTicketCounts.filter(x => x.countNo == 2);
         if (foundViolationTicketCount2.length > 0) {
           if (!foundViolationTicketCount2[0].description) foundViolationTicketCount2[0].description = ocrViolationTicket.fields["counts.count_no_2.description"].value;
-          if (!foundViolationTicketCount2[0].actOrRegulationNameCode) foundViolationTicketCount2[0].actOrRegulationNameCode = ocrViolationTicket.fields["counts.count_no_2.act_or_regulation_name_code"].value;
+          if (!foundViolationTicketCount2[0].actOrRegulationNameCode) foundViolationTicketCount2[0].actOrRegulationNameCode = ocrViolationTicket.fields["counts.count_no_1.act_or_regulation_name_code"] ? ocrViolationTicket.fields["counts.count_no_1.act_or_regulation_name_code"].value : "";
```
Comment (Sharkww) [inline src/frontend/staff-portal/src/app/services/violation-ticket.service.ts:132]: count_no_2

File: src/frontend/staff-portal/src/app/services/violation-ticket.service.ts [hunk 2]
```diff
@@ -138,7 +138,7 @@ export class ViolationTicketService implements IViolationTicketService {
           let violationTicketCount = {
             countNo: 2,
             description: ocrViolationTicket.fields["counts.count_no_2.description"].value,
-            actOrRegulationNameCode: ocrViolationTicket.fields["counts.count_no_2.act_or_regulation_name_code"].value,
+            actOrRegulationNameCode: ocrViolationTicket.fields["counts.count_no_1.act_or_regulation_name_code"] ? ocrViolationTicket.fields["counts.count_no_1.act_or_regulation_name_code"].value : "",
```
Comment (Sharkww) [inline src/frontend/staff-portal/src/app/services/violation-ticket.service.ts:141]: same as above

File: src/frontend/staff-portal/src/app/services/violation-ticket.service.ts [hunk 3]
```diff
@@ -164,7 +164,7 @@ export class ViolationTicketService implements IViolationTicketService {
         const foundViolationTicketCount3 = violationTicket.violationTicketCounts.filter(x => x.countNo == 3);
         if (foundViolationTicketCount3.length > 0) {
           if (!foundViolationTicketCount3[0].description) foundViolationTicketCount3[0].description = ocrViolationTicket.fields["counts.count_no_3.description"].value;
-          if (!foundViolationTicketCount3[0].actOrRegulationNameCode) foundViolationTicketCount3[0].actOrRegulationNameCode = ocrViolationTicket.fields["counts.count_no_3.act_or_regulation_name_code"].value;
+          if (!foundViolationTicketCount3[0].actOrRegulationNameCode) foundViolationTicketCount3[0].actOrRegulationNameCode = ocrViolationTicket.fields["counts.count_no_1.act_or_regulation_name_code"] ? ocrViolationTicket.fields["counts.count_no_1.act_or_regulation_name_code"].value : "";
```
Comment (Sharkww) [inline src/frontend/staff-portal/src/app/services/violation-ticket.service.ts:167]: same as above

File: src/frontend/staff-portal/src/app/services/violation-ticket.service.ts [hunk 4]
```diff
@@ -173,7 +173,7 @@ export class ViolationTicketService implements IViolationTicketService {
           let violationTicketCount = {
             countNo: 3,
             description: ocrViolationTicket.fields["counts.count_no_3.description"].value,
-            actOrRegulationNameCode: ocrViolationTicket.fields["counts.count_no_3.act_or_regulation_name_code"].value,
+            actOrRegulationNameCode: ocrViolationTicket.fields["counts.count_no_1.act_or_regulation_name_code"] ? ocrViolationTicket.fields["counts.count_no_1.act_or_regulation_name_code"].value : "",
```
Comment (Sharkww) [inline src/frontend/staff-portal/src/app/services/violation-ticket.service.ts:176]: same as above

**Review by Sharkww — CHANGES_REQUESTED** [review]
count number is not correct

File: src/frontend/staff-portal/src/app/services/violation-ticket.service.ts (hunk 2 above)
Comment (KDNeufeld) [reply src/frontend/staff-portal/src/app/services/violation-ticket.service.ts]: Excellent catch. Silly cut n' paste.

**Review by Sharkww — COMMENTED** [review]
Just asking which part was showing 'undefined' because it should be showing "" in frontend already if is null or undefined

**Comment by KDNeufeld** [comment]
> Just asking which part was showing 'undefined' because it should be showing "" in frontend already if is null or undefined

The `ocrViolationTicket.fields["counts.count_no_1.act_or_regulation_name_code"]` field is undefined for VT2 images so the `.value` was throwing an error.

File: src/frontend/staff-portal/src/app/services/violation-ticket.service.ts (hunk 1 above)
Comment (KDNeufeld) [reply src/frontend/staff-portal/src/app/services/violation-ticket.service.ts]: Excellent catch. Silly cut n' paste.

File: src/frontend/staff-portal/src/app/services/violation-ticket.service.ts (hunk 3 above)
Comment (KDNeufeld) [reply src/frontend/staff-portal/src/app/services/violation-ticket.service.ts]: Excellent catch. Silly cut n' paste.

File: src/frontend/staff-portal/src/app/services/violation-ticket.service.ts (hunk 4 above)
Comment (KDNeufeld) [reply src/frontend/staff-portal/src/app/services/violation-ticket.service.ts]: Excellent catch. Silly cut n' paste.

**Comment by Sharkww** [comment]
ok, can you try `ocrViolationTicket.fields["counts.count_no_1.act_or_regulation_name_code"]?.value` ? Thanks

**Comment by KDNeufeld** [comment]
> ok, can you try `ocrViolationTicket.fields["counts.count_no_1.act_or_regulation_name_code"]?.value` ? Thanks

That's much cleaner, thanks. Done.

**Review by Sharkww — APPROVED** [review]

**Pull request merged**